### ISSUE TITLE
Vimdoc parser without `PR#134` causes `heading=nil`

### ISSUE THREAD

**Issue by edte** [OXY2DEV/helpview.nvim#4]
when exec `h fold-manual` it failed

```
Error executing lua callback: ...site/pack/lazy/opt/helpview.nvim/lua/helpview/parser.lua:68: attempt to index local 'heading' (a nil value)
stack traceback:
        ...site/pack/lazy/opt/helpview.nvim/lua/helpview/parser.lua:68: in function 'vimdoc'
        ...site/pack/lazy/opt/helpview.nvim/lua/helpview/parser.lua:359: in function 'fn'
        ...6/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:473: in function 'for_each_tree'
        ...site/pack/lazy/opt/helpview.nvim/lua/helpview/parser.lua:355: in function 'init'
        ...arvim/site/pack/lazy/opt/helpview.nvim/ftplugin/help.lua:75: in function <...arvim/site/pack/lazy/opt/helpview.nvim/ftplugin/help.lua:32>
```

**Comment by OXY2DEV** [comment]
I am not getting any errors though?

Can you send a screenshot?

**Comment by taiwithers** [comment]
Here is my (debugging) init.lua with the error that occurs when I do `:h` (or any other help command). I'm not using any particular plugin manager, just using Nix/Home Manager to place the plugin downloads the vim path.

![image](https://github.com/user-attachments/assets/3dd5b7aa-0b4b-4840-a865-1c66c74bcee5)

**Comment by OXY2DEV** [comment]
> However using a modified version of the minimal init.lua provided by noice, I am able to open help files without any issues and helpview is active.

Are you sure you don't have some other plugin that works on help files?

**Comment by taiwithers** [comment]
> Are you sure you don't have some other plugin that works on help files?

It doesn't appear to be that - running nvim with the following init.lua still generates the same error, and presumably no other plugins are being loaded (using `nvim -u helpview-init-nolazy.lua`)
```lua
require('helpview').setup()
```

**Comment by OXY2DEV** [comment]
It only happens on `:h fold-manual`, right?

**Comment by taiwithers** [comment]
> It only happens on `:h fold-manual`, right?

Any help command will cause the error for me.

**Comment by OXY2DEV** [comment]
That is so strange. Even with the minimal `init.lua` I still can't reproduce the issue.

What neovim version are you using?

**Comment by taiwithers** [comment]
> What neovim version are you using?

![image](https://github.com/user-attachments/assets/af37c59c-9d2b-49fd-a228-7918f8970df1)
 
> Also did you update the plugin?

I'm using commit 336d731

I don't have any lua experience so apologies if I can't provide much assistance here.

**Comment by taiwithers** [comment]
> Can you manually add `vim.print(row_start)` before [this line](https://github.com/OXY2DEV/helpview.nvim/blob/main/lua/helpview/parser.lua#L68) and see what gets printed?

![image](https://github.com/user-attachments/assets/be527f3c-f488-456c-8be4-19c9a7fa97f7)

**Comment by OXY2DEV** [comment]
Which help file did you open?

> I need it to check what is on that line.

**Comment by taiwithers** [comment]
> Which help file did you open?
In the above screenshot, I was doing `:h` so it gives 40, 43, 60
`:h nvim` gives the rows 21, 72
`:h fold-manual` gives the rows 13, 273, 463, 576, 617

**Comment by OXY2DEV** [comment]
That is very strange.

![Screenshot_2024-08-18-08-10-43-470_com termux-edit](https://github.com/user-attachments/assets/1ea57d1e-fcc2-4006-a5c7-cccb18c85975)

What does `:InspectTree` show on these lines? Does `NixOs` get a different version of `tree-sitter`?

Anyway, you can just do `:h`, `40j` & run `:InspectTree` and see if it looks any different than my screenshot.

**Comment by taiwithers** [comment]
`:TSUpdate` says all parsers are up to date

Here's a screen recording of running that set of commands and it does land on an h1 element, though I have no idea why unless we're somehow looking at different help files. 

https://github.com/user-attachments/assets/1e36683d-a619-4572-98f4-6a7d06290539

**Comment by OXY2DEV** [comment]
What does :InspectTree show on those lines?

> Just do :h, 40j & run :InspectTree and see if it looks any different than my screenshot.

**Comment by OXY2DEV** [comment]
I have found the issue.

You treesitter shows `(word) (word)`. It should be showing `(delimiter) (heading)`.

**Comment by OXY2DEV** [comment]
There's definitely something wrong with the parser you are using (headings should be highlighted even without the plugin, which isn't the case in the video).

**Comment by OXY2DEV** [comment]
You most likely don't have [this PR](https://github.com/neovim/tree-sitter-vimdoc/pull/134) in your parser.

**Comment by taiwithers** [comment]
Yup, nixpkgs has nvim-treesitter rev 8012b55, which pulls vimdoc b711df (as per their [lockfile.json](https://github.com/nvim-treesitter/nvim-treesitter/blob/8012b55eee65eba1d1ee4df0a186d30e72dcbe65/lockfile.json#L798)), one commit before that PR is merged. So looks like myself and anyone else having issues with this need to be on at least nvim-treesitter rev 69341a2.
